lib/nrf_modem_lib/shell/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+target_include_directories(app PRIVATE .)
+target_sources_ifdef(CONFIG_NRF_MODEM_LIB_SHELL_TRACE app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/trace.c)
```

Comment:
I don't think it makes a difference, but I would use this instead:
`zephyr_library_sources_ifdef(CONFIG_NRF_MODEM_LIB_SHELL_TRACE app PRIVATE trace.c)`
Path should be relative when using this function, so passing just the C file name is fine.